django/views/generic/edit.py

```diff
@@ -112,7 +112,15 @@ def form_invalid(self, form):
         If the form is invalid, re-render the context data with the
         data-filled form and errors.
         """
-        return self.render_to_response(self.get_context_data(form=form))
+        return self.render_to_response(self.get_context_data())
+
+    def get_context_data(self, **kwargs):
+        """
+        Insert the form into the context dict.
+        """
+        context = {'form': self.get_form()}
+        context.update(kwargs)
```

Comment:
This looks like a good candidate for `kwargs.setdefault('form', self.get_form())`